packages/backend-app-api/src/services/implementations/httpRouter/middleware/cookieMiddlewareFactory.ts

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2024 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { getBearerTokenFromAuthorizationHeader } from '@backstage/plugin-auth-node';
+import { NextFunction, Request, Response } from 'express';
+import { decodeJwt } from 'jose';
+import { URL } from 'url';
+import lzstring from 'lz-string';
+import { Config } from '@backstage/config';
+
+function setTokenCookie(
+  res: Response,
+  options: { token: string; secure: boolean; cookieDomain: string },
+) {
+  try {
+    const payload = decodeJwt(options.token);
+    res.cookie('token', options.token, {
```

Comment:
Is it worth encrypting the token here so that only the server that issued the cookie can decrypt and retrieve token from the cookie?

Reply:
I don't think so? They would already have access to the bearer token that is being sent elsewhere. If they were to change the token or cookie, the JWT is signed and modifications would be caught by the default implementations of identityApi and tokenManager